js/packages/seed-vault/src/types.ts

```diff
@@ -0,0 +1,91 @@
+type AuthToken = number;
+
+type Base64EncodedAddress = string;
+
+type Base64EncodedSignature = string;
+
+type Base64EncodedPayload = string;
+
+type Base64EncodedMessage = Base64EncodedPayload;
+
+type Base64EncodedTransaction = Base64EncodedPayload;
+
+type DerivationPath = string;
+
+export type Account = Readonly<{
+    id: number,
+    name: string,
+    derivationPath: DerivationPath,
+    publicKeyEncoded: Base64EncodedAddress
+}>;
+
+export const SeedPurpose = {
+    SignSolanaTransaction: 0,
+} as const
+export type SeedPurpose = typeof SeedPurpose[keyof typeof SeedPurpose]
+
+export type Seed = Readonly<{
+    authToken: AuthToken,
+    name: string,
+    purpose: SeedPurpose
+}>;
+
+export type SeedPublicKey = Readonly<{
+    publicKey: Uint8Array,
+    publicKeyEncoded: Base64EncodedAddress,
+    resolvedDerivationPath: DerivationPath
+}>
+
+export type SigningRequest = Readonly<{
+    payload: Base64EncodedPayload,
+    requestedSignatures: [DerivationPath]
+}>;
+
+export type SigningResult = Readonly<{
+    signatures: [Base64EncodedSignature],
+    resolvedDerivationPaths: [DerivationPath]
+}>;
+
+interface AuthorizeSeedAPI {
+    hasUnauthorizedSeeds(): boolean
+    getAuthorizedSeeds(): [Seed]
+    authorizeNewSeed(): {authToken: AuthToken}
+    deauthorizeSeed(authToken: AuthToken): void
+}
+
+interface AccountAPI {
+    getAccounts(authToken: AuthToken): [Account]
+    updateAccountName(authToken: AuthToken, accountId: number, name?: string): void
+}
+
+interface CreateNewSeedAPI {
+    createNewSeed(): {authToken: AuthToken}
+}
+
+interface ImportExistingSeedAPI {
+    importExistingSeed(): {authToken: AuthToken}
+}
+
+interface PublicKeyAPI {
+    getPublicKey(authToken: AuthToken, derivationPath: DerivationPath): [SeedPublicKey]
+    getPublicKeys(authToken: AuthToken, derivationPaths: [DerivationPath]): [SeedPublicKey]
```

Comment:
suggestion: Pass a single object as a param, with each actual param as keys for each API method.

```
getPublicKeys(params: { authToken: AuthToken, derivationPaths: [DerivationPath] })
signMessages(params: { authToken: AuthToken, signingRequests: [SigningRequest] })
```

Can help with readability and reduces the chances of inputting the wrong parameter. The [MWA wallet API types ](https://github.com/solana-mobile/mobile-wallet-adapter/blob/main/js/packages/mobile-wallet-adapter-protocol/src/types.ts#L130) follow the same idea

Reply:
> Can help with readability and reduces the chances of inputting the wrong parameter.
Can you explain this a bit more? a params object is much less readable than straight up input arguments. And regarding inputting the wrong parameters... how is an object less prone to this? the arguments are typed. the IDE will throw an error if you don't pass a string for derivationPath, for example.  A params object is prone to serialization errors on the native side where clearly typed arguments are not.

Reply:
>  And regarding inputting the wrong parameters... how is an object less prone to this?

Those examples I chose are bad actually. Its more helpful in a case when the parameters are all the same type.

ie: for signMessage its 3 string params in a row.
```
signMessage({ authToken: "abc", derivationPath: "xyz", message: "message" })
// vs
signMessage("abc", "xyz", "message")
```

The object parameter clearly shows to the user what each slot in the param list represents so lessens the chance of confusing p1 with p2, etc.

> A params object is prone to serialization errors on the native side where clearly typed arguments are not.
In terms of type safety, both approaches should be equivalent I think. You can also abstract it and unwrap the object into individual params if it helps with serialization on the native side.